### DISCUSSION TITLE
How to apply custom RLS?

### DISCUSSION THREAD

**Discussion by nicitaacom** [supabase/supabase#18895] [Questions]
I want to apply this RLS policy 
```sql
CREATE POLICY "allow select for !user and user based on their ids" ON "public"."messages"
AS PERMISSIVE FOR SELECT
TO public
USING (sender_id = current_setting('app.anonymousId', true)::uuid OR sender_id = auth.uid())
```

I want allow in my RLS to select messages for anonymous users based on their's anonymousId value in cookies 
ALSO
I want to allow select access for authenticated users based on their ids

My table definition
```sql
create table
  public.messages (
    created_at timestamp with time zone not null default now(),
    conversation_id uuid not null,
    body text not null,
    sender_id text not null,
    is_closed_conversation boolean not null default false,
    id uuid not null default gen_random_uuid (),
    constraint messages_pkey primary key (id)
  ) tablespace pg_default;
```

I got this error
```
Error adding policy: failed to create pg.policies: operator does not exist: text = uuid
```

How to fix this error to apply RLS policy

**Comment by steve-chavez** [comment]
Check the error, the `=` operator is only defined for operands of the same type.

`sender_id` is `text` and `current_setting('app.anonymousId', true)::uuid` is `uuid`.

Make `sender_id` an `uuid` or stop casting the `current_setting`.

**Reply by nicitaacom** [reply]
If I run this
```sql
CREATE POLICY "allow select for !user and user based on their ids" ON "public"."messages"
AS PERMISSIVE FOR SELECT
TO public
USING (sender_id = current_setting('app.anonymousId', true)::uuid OR sender_id::uuid = auth.uid())
```

I got the same error

If I change type for column `sender_id` to `uuid` - I got the same error

**Reply by nicitaacom** [reply]
Its not the answer because it throws the same error
